pkg/template/core/starlark_struct.go

```diff
@@ -5,17 +5,22 @@ package core
 
 import (
 	"fmt"
-
 	"github.com/k14s/starlark-go/starlark"
 	"github.com/k14s/ytt/pkg/orderedmap"
 )
 
 type StarlarkStruct struct {
-	data *orderedmap.Map
+	data *orderedmap.Map // [string]starlark.Value; most common usage: HasAttrs
+}
+
+func NewStarlarkStruct(goStringKeyToStarlarkValue *orderedmap.Map) *StarlarkStruct {
+	return &StarlarkStruct{data: goStringKeyToStarlarkValue}
 }
 
-var _ starlark.Value = &StarlarkStruct{}
-var _ starlark.HasAttrs = &StarlarkStruct{}
+var _ starlark.Value = (*StarlarkStruct)(nil)
+var _ starlark.HasAttrs = (*StarlarkStruct)(nil)
+var _ starlark.IterableMapping = (*StarlarkStruct)(nil)
+var _ starlark.Sequence = (*StarlarkStruct)(nil)
 
 func (s *StarlarkStruct) String() string        { return "struct(...)" }
 func (s *StarlarkStruct) Type() string          { return "struct" }
```

Comment:
it's interesting that Truth() method always returns true in starlark's official struct: https://github.com/google/starlark-go/blob/master/starlarkstruct/struct.go#L159. what other difference are note worthy? hashable?

Reply:
another interesting difference is return of NoSuchAttrError (https://github.com/google/starlark-go/blob/master/starlarkstruct/struct.go#L231) which makes runtime augment err with a hint of misspellings.

Reply:
> it's interesting that Truth() method always returns true in Starlark's official struct

seriously.  Yeah, and _with a comment_ that reads, `// even when empty`. Gonna chew on what "truth" means, here.

Reply:
With respect to _additional_ functionality...

I _am_ interested in pulling some of that over (love that hinting bit, yes).

That said, given that we were basically using the `ytt` version of `struct` for most of our own cases, I feel less compelled to tackle that in _this_ PR.

I _did_ review `yamlfragment` as #208 requested.

Seems "good enough" for now.

Reply:
Truth:

So... all these data structures are truthy if only if they are non-empty.
```
func (d *Dict) Truth() Bool { return d.Len() > 0 }
func (l *List) Truth() Bool { return l.Len() > 0 }
func (s *Set) Truth() Bool { return s.Len() > 0 }
func (t Tuple) Truth() Bool { return len(t) > 0 }
func (r rangeValue) Truth() Bool { return r.len > 0 }
```

why would `struct` be different? 🤔

Reply:
not sure why they have done that. im ok keeping it inline especially that yamlfrag also does len() > 0.